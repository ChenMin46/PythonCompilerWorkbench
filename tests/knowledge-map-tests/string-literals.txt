"hello"
'hello'
"This is a sentence."
"John's cat."
'He liked watching "The Matrix"!'
""
" "
"\t \\ \" \'"
"John's\n\n\n\ndog."
