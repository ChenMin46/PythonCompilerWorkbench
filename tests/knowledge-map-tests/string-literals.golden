"hello"
'hello'
"This is a sentence."
"John's cat."
'He liked watching "The Matrix"!'
""
" "
"\t \\ \" \'"
"John's\n\n\n\ndog."

",".join([1,2,3])
===
Module
  body: [
    Expr
      value: Str >> '"hello"'
        s: 'hello'
    Expr
      value: Str >> "'hello'"
        s: 'hello'
    Expr
      value: Str >> '"This is a sentence."'
        s: 'This is a sentence.'
    Expr
      value: Str >> '"John\'s cat."'
        s: "John's cat."
    Expr
      value: Str >> '\'He liked watching "The Matrix"!\''
        s: 'He liked watching "The Matrix"!'
    Expr
      value: Str >> '""'
        s: ''
    Expr
      value: Str >> '" "'
        s: ' '
    Expr
      value: Str >> '"\\t \\\\ \\" \\\''
        s: '\t \\ " \''
    Expr
      value: Str >> '"John\'s\\n\\n\\n\\ndog."'
        s: "John's\n\n\n\ndog."
    Expr
      value: Call >> '",".join([1,2,3])'
        func: Attribute >> '",".join'
          value: Str >> '","'
            s: ','
          attr: 'join'
          ctx: Load
        args: [
          List >> '[1,2,3]'
            elts: [
              Num >> '1'
                n: 1
              Num >> '2'
                n: 2
              Num >> '3'
                n: 3
            ]
            ctx: Load
        ]
        keywords: [
        ]
        starargs: None
        kwargs: None
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "contents": [
                "\"hello\""
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "'hello'"
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "\"This is a sentence.\""
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "\"John's cat.\""
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "'He liked watching \"The Matrix\"!'"
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "\"\""
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "\" \""
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n", 
        {
          "contents": [
            {
              "contents": [
                "\"\\t \\\\ \\\" \\'"
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\"\n", 
        {
          "contents": [
            {
              "contents": [
                "\"John's\\n\\n\\n\\ndog.\""
              ], 
              "id": "id_0", 
              "name": "Str"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }, 
        "\n\n", 
        {
          "contents": [
            {
              "contents": [
                {
                  "contents": [
                    {
                      "contents": [
                        "\",\""
                      ], 
                      "id": "id_0", 
                      "name": "Str"
                    }, 
                    ".join"
                  ], 
                  "id": "id_0", 
                  "name": "Attribute"
                }, 
                {
                  "id": "tid_1", 
                  "name": "token", 
                  "value": "("
                }, 
                {
                  "contents": [
                    {
                      "id": "tid_2", 
                      "name": "token", 
                      "value": "["
                    }, 
                    {
                      "contents": [
                        "1"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_3", 
                      "name": "token", 
                      "value": ","
                    }, 
                    {
                      "contents": [
                        "2"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_4", 
                      "name": "token", 
                      "value": ","
                    }, 
                    {
                      "contents": [
                        "3"
                      ], 
                      "id": "id_0", 
                      "name": "Num"
                    }, 
                    {
                      "id": "tid_5", 
                      "name": "token", 
                      "value": "]"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "List"
                }, 
                {
                  "id": "tid_6", 
                  "name": "token", 
                  "value": ")"
                }
              ], 
              "id": "id_0", 
              "name": "Call"
            }
          ], 
          "id": "id_0", 
          "name": "Expr"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
